### DISCUSSION TITLE
Adding new interface so-monitor-add error: master='bond0' doesn't refer to any existing profile

### DISCUSSION THREAD

**Discussion by jethrogillgren** [Security-Onion-Solutions/securityonion#6242] [Unsupported Versions] [answered]
Hi, we have an existing Security Onion 2 instance with a single monitor interface, but now need to add a second interface.
We have configured the extra interface (eth2) the same as the first (eth1), and found the docs [here](https://docs.securityonion.net/en/2.3/so-monitor-add.html) for running `so-monitor-add`.

We get an error we don't understand, is anyone able to shed some light on the issue?

```
[onion@security-onion ~]$ sudo so-monitor-add eth2
Warning: master='bond0' doesn't refer to any existing profile.
Error: Failed to add 'bond0-slave-eth2' connection: connection.slave-type: Cannot set 'master' without 'slave-type'
```

It occurs on staging and production. Persists after reboot. The new interface is configured the same as the old interface. The old interface is still monitoring traffic OK.

Thanks for any help.

**Comment by dougburks** [comment]
What is the output of the following?
```
ifconfig bond0
```

**Reply by jethrogillgren** [reply]
I get `bond0: error fetching interface information: Device not found`

The reason I said the old interface is still working was because I could see monitor traffic in grafana:

![Screenshot 2021-11-18 at 14 13 30](https://user-images.githubusercontent.com/2091292/142431558-a79953b8-6a1f-4d4c-bbe2-2ca807b0a349.jpg)

and `so-status` says everything is `ok`.

But sounds like I'm missing something important?

**Reply by dougburks** [reply]
Yes, by default, Security Onion should add your monitoring interface to `bond0` and then configure all processes to sniff traffic from `bond0`.

How is Suricata sniffing traffic?
```
ps aux |grep suricata |grep af-packet
```

**Reply by jethrogillgren** [reply]
`suricata 22013 10.3  1.6 964996 551200 ?       Sl   07:03  33:46 /opt/suricata/bin/suricata -c /etc/suricata/suricata.yaml --af-packet=eth1 --user=940 --group=940 --pidfile /var/run/suricata.pid -F /etc/suricata/bpf`

For info: we installed on AWS using the AMI, and are on verison `2.3.80`.

**Reply by dougburks** [reply]
Ahh yes, cloud images do not use `bond0`. I've updated the documentation with a warning:
https://docs.securityonion.net/en/2.3/so-monitor-add.html

**Reply by jethrogillgren** [reply]
Aah apoligies!
Do you know if there is a way to add new monitor interfaces on the cloud image?
When using the image with AWS Traffic Mirroring you get a limit of 10 sources going to each interface.
If it's non-trivial to add monitor interfaces on the cloud AMI, we will re-create and just give enough interfaces for the forseable future.

**Comment by weslambert** [comment, marked as answer]
The recommendation here would be to use a NLB (Network Load Balancer) to send to the monitor interface, as opposed to setting up multiple monitor interfaces. We don't currently support the usage of multiple monitor interfaces in AWS, due to the fact that a bonded monitor interface cannot be used (AWS limitation).

**Reply by jethrogillgren** [reply]
Thank-you, we will move to a NLB.
